lib/_stream_wrap.js

```diff
@@ -1,223 +1,3 @@
-  this.stream.pause();
-  return 0;
-};
-
-StreamWrap.prototype.doShutdown = function doShutdown(req) {
-  const self = this;
-  const handle = this._handle;
-  const item = this._enqueue('shutdown', req);
-
-  this.stream.end(function() {
-    // Ensure that write was dispatched
-    setImmediate(function() {
-      if (!self._dequeue(item))
-        return;
-
-      handle.finishShutdown(req, 0);
-    });
-  });
-  return 0;
-};
-
-StreamWrap.prototype.doWrite = function doWrite(req, bufs) {
-  const self = this;
-  const handle = self._handle;
-
-  var pending = bufs.length;
-
-  // Queue the request to be able to cancel it
-  const item = self._enqueue('write', req);
-
-  self.stream.cork();
-  for (var n = 0; n < bufs.length; n++)
-    self.stream.write(bufs[n], done);
-  self.stream.uncork();
-
-  function done(err) {
-    if (!err && --pending !== 0)
-      return;
-
-    // Ensure that this is called once in case of error
-    pending = 0;
-
-    let errCode = 0;
-    if (err) {
-      const code = uv[`UV_${err.code}`];
-      errCode = (err.code && code) ? code : uv.UV_EPIPE;
-    }
-
-    // Ensure that write was dispatched
-    setImmediate(function() {
-      // Do not invoke callback twice
-      if (!self._dequeue(item))
-        return;
-
-      handle.doAfterWrite(req);
-      handle.finishWrite(req, errCode);
-    });
-  }
-
-  return 0;
-};
-
-function QueueItem(type, req) {
-  this.type = type;
-  this.req = req;
-  this.prev = this;
-  this.next = this;
-}
-
-StreamWrap.prototype._enqueue = function _enqueue(type, req) {
-  const item = new QueueItem(type, req);
-  if (this._list === null) {
-    this._list = item;
-    return item;
-  }
-
-  item.next = this._list.next;
-  item.prev = this._list;
-  item.next.prev = item;
-  item.prev.next = item;
-
-  return item;
-};
-
-StreamWrap.prototype._dequeue = function _dequeue(item) {
-  assert(item instanceof QueueItem);
-
-  var next = item.next;
-  var prev = item.prev;
-
-  if (next === null && prev === null)
-    return false;
-
-  item.next = null;
-  item.prev = null;
-
-  if (next === item) {
-    prev = null;
-    next = null;
-  } else {
-    prev.next = next;
-    next.prev = prev;
-  }
-
-  if (this._list === item)
-    this._list = next;
-
-  return true;
-};
-
-StreamWrap.prototype.doClose = function doClose(cb) {
-  const self = this;
-  const handle = self._handle;
-
-  setImmediate(function() {
-    while (self._list !== null) {
-      const item = self._list;
-      const req = item.req;
-      self._dequeue(item);
-
-      const errCode = uv.UV_ECANCELED;
-      if (item.type === 'write') {
-        handle.doAfterWrite(req);
-        handle.finishWrite(req, errCode);
-      } else if (item.type === 'shutdown') {
-        handle.finishShutdown(req, errCode);
-      }
-    }
-
-    // Should be already set by net.js
-    assert(self._handle === null);
-    cb();
-  });
-};
+module.exports = require('internal/wrap_js_stream');
```

Comment:
I would deprecate this file, people should not be requiring this in any form.

Reply:
**If** we go with semver-major, yes.